sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobsStorageResourceProvider.cs

```diff
@@ -43,6 +43,25 @@ private enum CredentialType
         private readonly Func<Uri, CancellationToken, ValueTask<StorageSharedKeyCredential>> _getStorageSharedKeyCredential;
         private readonly Func<Uri, CancellationToken, ValueTask<AzureSasCredential>> _getAzureSasCredential;
 
+        /// <summary>
+        /// <para>
+        /// Constructs this provider to use no credentials when making a new Blob Storage
+        /// <see cref="StorageResource"/>.
+        /// </para>
+        /// <para>
+        /// This instance will NOT use any credential when constructing the underlying
+        /// Azure.Storage.Blobs client, e.g. <see cref="BlockBlobClient(Uri, BlobClientOptions)"/>.
+        /// This is for the purpose of either anonymous access when constructing the client. When
+        /// constructing from a pre-existing client (e.g.
+        /// <see cref="FromClient(BlockBlobClient, BlockBlobStorageResourceOptions)"/>),
+        /// the client's credentials will be used.
```

Comment:
Nit: do we want to remove the bits about `FromClient` now since they are static and so you don't need to construct this?